CONTRIBUTING.md

```diff
@@ -1,51 +1,56 @@
 
 ### Config
 
-A *config* knows about the code style rules and the files and directories that
-must be scanned by the tool when run in the directory of your project. It is
-useful for projects that follow a well-known directory structures (like for
-Symfony projects for instance).
-
-### How-To
-
-* [Fork](https://help.github.com/articles/fork-a-repo/) the repo.
-* [Checkout](https://git-scm.com/docs/git-checkout) the branch you want to make changes on:
-  * If you are fixing a bug or typo, improving tests or for any small tweak: the lowest branch where the changes can be applied. Once your Pull Request is accepted, the changes will get merged up to highest branches.
-  * `master` in other cases (new feature, deprecation, or backwards compatibility breaking changes). Note that most of the time, `master` represents the next minor release of PHP CS Fixer, so Pull Requests that break backwards compatibility might be postponed.
-* Install dependencies: `composer install`.
-* Create a new branch, e.g. `feature-foo` or `bugfix-bar`.
-* Make changes.
-* If you are adding functionality or fixing a bug - add a test! Prefer adding new test cases over modifying existing ones.
-* Make sure there is dev environment is correct: `composer dev-tools:check`.
+A *config* knows about the code style rules and the files and directories that must be scanned by the tool when run in the context of your project. It is useful for projects that follow a well-known directory structures, but the tool is not limited to any specific structure, and you can configure it in a very flexible way.
+
+## How to contribute
+
+### Development
+
+* [Fork](https://help.github.com/articles/fork-a-repo/) the repo. You can use native Git approach or use [`gh` CLI tool](https://cli.github.com/).
+* Create new branch on top of the latest revision of `master` branch (if you already had project locally, then make sure to update this branch before going to next steps). It's good when branch's name reflects intent of the changes, but this is not strict requirement since Pull Request provides description of the change. However, with good branch naming it's easier to work on multiple changes simultaneously.
+* Install dependencies by running `composer update` (since project does not contain `composer.lock` it's better to ensure latest versions of packages by running `update` command instead of `install`).
+* Make changes. Please remember that **all** changes have to be covered with tests.
+  * if you work on a bug fix, please start with reproducing the problem by adding failing test case(s). When you have failing test case(s), you can [create Pull Request](#opening-a-pull-request) just to reproduce fail in the CI. Then you can provide fix _in the subsequent commits_, it will make code review easier. It's allowed to modify existing test cases in bug fix Pull Request, but *only if* current behavior is proved to be invalid.
```

Comment:
```suggestion
  * if you work on a bug fix, please start with reproducing the problem by adding failing test case(s). When you have failing test case(s), you can [create pull request](#opening-a-pull-request) just to reproduce fail in the CI. Then you can provide fix _in the subsequent commits_, it will make code review easier. It's allowed to modify existing test cases in bug fix pull request, but *only if* current behavior is proved to be invalid.
```